homeassistant/components/econet/sensor.py

```diff
@@ -1,113 +1,124 @@
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from . import EcoNetEntity
 from .const import DOMAIN, EQUIPMENT
 
-ENERGY_KILO_BRITISH_THERMAL_UNIT = "kBtu"
-
-TANK_HEALTH = "tank_health"
-AVAILABLE_HOT_WATER = "available_hot_water"
-COMPRESSOR_HEALTH = "compressor_health"
-OVERRIDE_STATUS = "override_status"
-WATER_USAGE_TODAY = "water_usage_today"
-POWER_USAGE_TODAY = "power_usage_today"
-ALERT_COUNT = "alert_count"
-WIFI_SIGNAL = "wifi_signal"
-RUNNING_STATE = "running_state"
-
-SENSOR_NAMES_TO_ATTRIBUTES = {
-    TANK_HEALTH: "tank_health",
-    AVAILABLE_HOT_WATER: "tank_hot_water_availability",
-    COMPRESSOR_HEALTH: "compressor_health",
-    OVERRIDE_STATUS: "override_status",
-    WATER_USAGE_TODAY: "todays_water_usage",
-    POWER_USAGE_TODAY: "todays_energy_usage",
-    ALERT_COUNT: "alert_count",
-    WIFI_SIGNAL: "wifi_signal",
-    RUNNING_STATE: "running_state",
-}
-
-SENSOR_NAMES_TO_UNIT_OF_MEASUREMENT = {
-    TANK_HEALTH: PERCENTAGE,
-    AVAILABLE_HOT_WATER: PERCENTAGE,
-    COMPRESSOR_HEALTH: PERCENTAGE,
-    OVERRIDE_STATUS: None,
-    WATER_USAGE_TODAY: UnitOfVolume.GALLONS,
-    POWER_USAGE_TODAY: None,  # Depends on unit type
-    ALERT_COUNT: None,
-    WIFI_SIGNAL: None,
-    RUNNING_STATE: None,  # This is just a string
-}
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="tank_health",
+        name="tank_health",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="tank_hot_water_availability",
+        name="available_hot_water",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="compressor_health",
+        name="compressor_health",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="override_status",
+        name="override_status",
+    ),
+    SensorEntityDescription(
+        key="todays_water_usage",
+        name="water_usage_today",
+        native_unit_of_measurement=UnitOfVolume.GALLONS,
+        device_class=SensorDeviceClass.WATER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="todays_energy_usage",
+        name="power_usage_today",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="alert_count",
+        name="alert_count",
+    ),
+    SensorEntityDescription(
+        key="wifi_signal",
+        name="wifi_signal",
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
```

Comment:
Signal strength sensors are usually disabled by default
```suggestion
        entity_registry_enabled_default=False,
        entity_category=EntityCategory.DIAGNOSTIC,
    ),
```

Reply:
I think your suggestion is on the wrong line. Is the way it is now correct? Adding:
```
entity_registry_enabled_default=False,
entity_category=EntityCategory.DIAGNOSTIC,
```
To the signal strength sensor entity description?